src/viewProviders/flinkStatements.ts

```diff
@@ -1,63 +1,83 @@
-import { TreeDataProvider, TreeItem } from "vscode";
+import { TreeDataProvider, TreeItem, window } from "vscode";
 import { ContextValues } from "../context/values";
-import { currentFlinkStatementsPoolChanged } from "../emitters";
-import { CCloudFlinkComputePool, FlinkComputePool } from "../models/flinkComputePool";
+import { currentFlinkStatementsResourceChanged } from "../emitters";
+import { CCloudResourceLoader, ResourceLoader } from "../loaders";
+import { CCloudEnvironment } from "../models/environment";
+import { CCloudFlinkComputePool } from "../models/flinkComputePool";
 import { FlinkStatement, FlinkStatementTreeItem } from "../models/flinkStatement";
 import { BaseViewProvider } from "./base";
 
+/**
+ * View controller for Flink statements. Can be assigned to track either
+ * a single compute cluster, or a CCloud environment.
+ *
+ * If set to a CCloud environment, will show all of the statements
+ * across all of the provider+region pairs that we find Flinkable
+ * within the environment. See {@link CCloudResourceLoader.getFlinkStatements} for
+ * the specifics.
+ *
+ * */
 export class FlinkStatementsViewProvider
-  extends BaseViewProvider<FlinkComputePool, FlinkStatement>
+  extends BaseViewProvider<CCloudFlinkComputePool | CCloudEnvironment, FlinkStatement>
   implements TreeDataProvider<FlinkStatement>
 {
   loggerName = "viewProviders.flinkStatements";
   viewId = "confluent-flink-statements";
 
-  parentResourceChangedEmitter = currentFlinkStatementsPoolChanged;
+  parentResourceChangedEmitter = currentFlinkStatementsResourceChanged;
   parentResourceChangedContextValue = ContextValues.flinkStatementsPoolSelected;
 
   searchContextValue = ContextValues.flinkStatementsSearchApplied;
 
-  async getChildren(): Promise<FlinkStatement[]> {
-    const children: FlinkStatement[] = [];
-    if (!this.computePool) {
-      return children;
-    }
+  // Map of resource id string -> resource currently in the tree view.
+  private resourcesInTreeView: Map<string, FlinkStatement> = new Map();
```

Comment:
I could definitely see this being moved into the `BaseViewProvider` in a follow-on branch to help consolidate logic and help wrangle per-item updates 👍